osm2streets/src/lanes/mod.rs

```diff
@@ -355,3 +397,81 @@ impl fmt::Display for Direction {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum LtrLaneNum {
+    Forward(usize),
+    Backward(usize),
+}
+
+impl LtrLaneNum {
+    pub fn direction(&self) -> Direction {
+        match self {
+            Self::Forward(_) => Direction::Fwd,
+            Self::Backward(_) => Direction::Back,
+        }
+    }
+
+    pub fn number(&self) -> usize {
+        match self {
+            Self::Forward(num) | Self::Backward(num) => *num,
+        }
+    }
+
+    /// Converts to the same numbered lane in the opposite direction.
+    pub fn reverse(&self) -> Self {
+        use LtrLaneNum::*;
+        match self {
+            Forward(n) => Backward(*n),
+            Backward(n) => Forward(*n),
+        }
+    }
+}
+
+/// Identifies a position within the width of a roadway. Lanes are identified by their left-to-right
+/// position, as per the OSM convention.
```

Comment:
Early on I was tempted to represent this using `Inside`/`OutsideOf(i32)` naming lanes relative to the separation (with negative indicating lanes on the backward side) and adding an invariant to `Road` that it stores its separation. After working with the minutia of calculating things from the left edge using `lanes_ltr` I'm more comfortable with it - in big part because my understanding is that in the presence of contraflow lanes, `*:lanes:forward` tags, and thus `placement:forward`, refer to the forward lanes in lrt order, regardless of their position.